build/rules.mak

```diff
@@ -6,10 +6,20 @@ MODULE_FILES := \
                 $(addprefix $(MODULE_DIR)/,$(MODULE_FILES_TO_COPY)) \
                 $(addprefix $(MODULE_DIR)/,$(CUSTOM_TYPES_TO_COPY)) \
 
-
 RST_FILES := \
                 $(addprefix $(DRIVER_DOCS_DIR)/,$(RST_FILES_TO_GENERATE)) \
 
+EXAMPLE_FILES := $(if $(wildcard src/$(DRIVER)/examples/*),$(shell find src/$(DRIVER)/examples/* -type f -print),)
+
+# If there are any examples, we will need to build the examples zip file for this driver
+ifneq (,$(EXAMPLE_FILES))
+MKDIRECTORIES += $(ROOT_DIR)/examples
+
+DRIVER_EXAMPLES_ZIP_FILE := $(ROOT_DIR)/examples/$(DRIVER)_examples.zip
+MODULE_FILES += $(DRIVER_EXAMPLES_ZIP_FILE)
+
+endif # ifneq (,$(EXAMPLE_FILES))
+
```

Comment:
very minor, but for readability you could improve whitespace, i.e.

```
# If there are any examples, we will need to build the examples zip file for this driver
ifneq (,$(EXAMPLE_FILES))

MKDIRECTORIES += $(ROOT_DIR)/examples
DRIVER_EXAMPLES_ZIP_FILE := $(ROOT_DIR)/examples/$(DRIVER)_examples.zip
MODULE_FILES += $(DRIVER_EXAMPLES_ZIP_FILE)
 
endif # ifneq (,$(EXAMPLE_FILES))
```

or

```
# If there are any examples, we will need to build the examples zip file for this driver
ifneq (,$(EXAMPLE_FILES))
   MKDIRECTORIES += $(ROOT_DIR)/examples
   DRIVER_EXAMPLES_ZIP_FILE := $(ROOT_DIR)/examples/$(DRIVER)_examples.zip
   MODULE_FILES += $(DRIVER_EXAMPLES_ZIP_FILE)
endif # ifneq (,$(EXAMPLE_FILES))
```

or something like that

Reply:
Fixed